pkg/dns/config/sync.go

```diff
@@ -47,50 +36,28 @@ type Sync interface {
 	Periodic() <-chan *Config
 }
 
+type SyncResult struct {
+	Version string
+	Data    map[string]string
+}
+
+type SyncSource interface {
```

Comment:
source just gets the data (from the API or from the disk)

that's logically distinct from deduping on version, parsing and validating that data into a config